templates/k8s/job-build.jinja2

```diff
@@ -0,0 +1,71 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ job_name }}
+spec:
+  backoffLimit: 3
+  template:
+    spec:
+      restartPolicy: OnFailure
+
+      # in-memory tmpfs
+      volumes:
+      - name: scratch-volume
+        emptyDir: { medium: "Memory" }
+
+      # only run on preemptible nods
+      #nodeSelector:
+      #  cloud.google.com/gke-preemptible: "true"
+
+      containers:
+      - name: kci-build
+        image: {{ "FIXME" | env_override('DOCKER_BASE') }}build-{{ "FIXME" | env_override('DOCKER_IMAGE') }}
+
+        volumeMounts:
+        - mountPath: "/scratch"
+          name: scratch-volume
+
+        command: ["/bin/bash", "-x", "-c"]
+        args: ["echo nproc=$(nproc) && df && free && \
+export KDIR=/scratch/linux && export OUTPUT=$HOME/build && export CCACHE_DISABLE=true && env &&\
+git clone --depth=1 --branch ${KCI_CORE_BRANCH} ${KCI_CORE_URL} && cd kernelci-core && \
+./kci_build pull_tarball --kdir ${KDIR} --url ${SRC_TARBALL} --retries 3 --delete && \
+./kci_build build_kernel --kdir ${KDIR} --output ${OUTPUT} --defconfig=${DEFCONFIG} --arch=${ARCH} --build-env=${BUILD_ENVIRONMENT} --verbose && \
+./kci_build install_kernel --kdir ${KDIR} --output ${OUTPUT} --config ${BUILD_CONFIG} --describe=${GIT_DESCRIBE} --describe-verbose=${GIT_DESCRIBE_VERBOSE} --commit=${COMMIT_ID} && \
+./kci_build push_kernel --kdir ${KDIR} --token=${KCI_API_TOKEN} --api=${KCI_API_URL} && \
+./kci_build publish_kernel --kdir ${KDIR} --token=${KCI_API_TOKEN} --api=${KCI_API_URL} \
+"]
+
+        env:
+        - name: ARCH
+          value: "{{ "FIXME" | env_override('ARCH') }}"
+        - name: BUILD_ENVIRONMENT
+          value: "{{ "FIXME" | env_override('BUILD_ENVIRONMENT') }}"
+        - name: DEFCONFIG
+          value: "{{ "FIXME" | env_override('DEFCONFIG') }}"
+        - name: KCI_CORE_URL
+          value: "{{ "FIXME" | env_override('KCI_CORE_URL') }}"
+        - name: KCI_CORE_BRANCH
+          value: "{{ "FIXME" | env_override('KCI_CORE_BRANCH') }}"
+        - name: SRC_TARBALL
+          value: "{{ "FIXME" | env_override('SRC_TARBALL') }}"
+        - name: GIT_DESCRIBE
+          value: "{{ "FIXME" | env_override('GIT_DESCRIBE') }}"
+        - name: GIT_DESCRIBE_VERBOSE
+          value: "{{ "FIXME" | env_override('GIT_DESCRIBE_VERBOSE') }}"
+        - name: COMMIT_ID
+          value: "{{ "FIXME" | env_override('COMMIT_ID') }}"
+        - name: BUILD_CONFIG
+          value: "{{ "FIXME" | env_override('BUILD_CONFIG') }}"
+        - name: KCI_API_URL
+          value: "{{ "FIXME" | env_override('KCI_API_URL') }}"
+        - name: KCI_API_TOKEN
+          value: "{{ "FIXME" | env_override('KCI_API_TOKEN') }}"
+
+        resources:
+          limits:
+            cpu: "8"
```

Comment:
This probably wants to be joined up with the small/medium/big build stuff we've got.

Reply:
Yes, looks like it would be worth having a template parameter for this.

Reply:
Sure, I can parametrize this, but I think longer-term, we might want to rethink our small/medium big.  That's needed when we have a static pool of VM, and "big" builds would block others.  With k8s, we have clusters that can auto-scale based on demand, so we shouldn't need to manage the small/medium/large from the jenkins side, IMO.

Reply:
I think it might still be worth it from the point of view of helping the scheduler place things effectively, especially if we manage to do stuff like sharing the fetch of the source between multiple nodes - though from my fiddling with my stuff it might end up being more that we have things like separate templates for some of the builds, especially the smaller ones like most of the arm configs and allnoconfig.